tests/unit/pause.tcl

```diff
@@ -1,3 +1,11 @@
+proc wait_for_paused_clients_count {count {maxtries 100} {delay 10}} {
```

Comment:
maybe we should move this to `util.tcl`.
we keep adding local procs in one file, and then realizing a few months later that we want to use them in another file and moving them..

also, since we fail the test anyway, is there any disadvantage to add a CLIENT UNPAUSE in `wait_for_blocked_clients_count`? (maybe inside a catch)?

i.e. it's just an attempt to avoid a hung, and finish with error sooner.. i suppose it can't do any damage.

Reply:
> maybe we should move this to `util.tcl`. we keep adding local procs in one file, and then realizing a few months later that we want to use them in another file and moving them..

Yes

> also, since we fail the test anyway, is there any disadvantage to add a CLIENT UNPAUSE in `wait_for_blocked_clients_count`? (maybe inside a catch)?
> 
> i.e. it's just an attempt to avoid a hung, and finish with error sooner.. i suppose it can't do any damage.

1. I don't quite understand what you mean...I think the best way to express this kind of logic is "try-catch-finally", but it looks like we don't have it currently. We do have the `catch` expression, but using it without try and finally isn't enough for this case (at least I tried and it wasn't enough). Also sentences like `catch {e} {fail e}` look stupid...
Directly adding `UNPAUSE` in `wait_for_blocked_clients_count` may also OK, but I'd rather split these functions and not mix them for robustness.

2. If the test does something bad (like something that could lead to server abort) in pausing phase, the `UNPAUSE` command may introduce these bad commands into server. However we must execute `UNPAUSE` sooner or later (or wait for timeout), so I think it's better unpause before this single test returning `fail`.

Reply:
1. we can add another optional `code` argument to the `proc test` to be executed on failure, and then pass this cleanup code as an argument. then we'll need to modify lots of tests to use it instead of have the cleanup code when they're done. i don't think i wanna invest that effort now.
2. i didn't understand your argument, but i agree that at the current state we'd rather UNPAUE before `fail`. but since this is just a small effort to prevent the test suite from hanging after a failure, i think we can add it to the normal `wait_for_blocked_clients_count`, maybe inside a small `catch` and with a big comment explaining why we do that, rather than introduce a different method that semantically (on successful tests) does the same.
3. yes, as i said, when tests fail, there are many things not cleaned up. we're aware of that, and decided not to handle that at the moment.

Reply:
@oranagra Okay I got your ideas. As to this case, now I think the simplest fix is just shortening the `client PAUSE` duration from `100000000` to `10000` or lower. We won't have to bother commenting or changing these procs now :)